internal/commands/config/add_profile_internal.go

```diff
@@ -20,35 +20,43 @@ func RunInternalConfigAddProfile(rc io.ReadCloser) (err error) {
 		return fmt.Errorf("failed to add profile: %w", err)
 	}
 
-	err = profiles.GetMainConfig().ValidateNewProfileName(newProfileName)
+	err = profiles.GetKoanfConfig().ValidateNewProfileName(newProfileName)
 	if err != nil {
 		return fmt.Errorf("failed to add profile: %w", err)
 	}
 
 	output.Message(fmt.Sprintf("Adding new profile '%s'...", newProfileName), nil)
 
-	subViper := viper.New()
-	subViper.Set(options.ProfileDescriptionOption.ViperKey, newDescription)
+	subKoanf := koanf.New(".")
+	err = subKoanf.Set(options.ProfileDescriptionOption.KoanfKey, newDescription)
+	if err != nil {
+		return fmt.Errorf("failed to add profile: %w", err)
+	}
 
-	if err = profiles.GetMainConfig().SaveProfile(newProfileName, subViper); err != nil {
+	if err = profiles.GetKoanfConfig().SaveProfile(newProfileName, subKoanf); err != nil {
 		return fmt.Errorf("failed to add profile: %w", err)
 	}
 
-	output.Success(fmt.Sprintf("Profile created. Update additional profile attributes via 'pingcli config set' or directly within the config file at '%s'", profiles.GetMainConfig().ViperInstance().ConfigFileUsed()), nil)
+	output.Success(fmt.Sprintf("Profile created. Update additional profile attributes via 'pingcli config set' or directly within the config file at '%s'", profiles.GetKoanfConfig().GetKoanfConfigFile()), nil)
 
 	if setActive {
-		if err = profiles.GetMainConfig().ChangeActiveProfile(newProfileName); err != nil {
+		if err = profiles.GetKoanfConfig().ChangeActiveProfile(newProfileName); err != nil {
 			return fmt.Errorf("failed to set active profile: %w", err)
 		}
 
 		output.Success(fmt.Sprintf("Profile '%s' set as active.", newProfileName), nil)
 	}
 
-	err = profiles.GetMainConfig().DefaultMissingViperKeys()
+	err = profiles.GetKoanfConfig().DefaultMissingKoanfKeys()
 	if err != nil {
 		return fmt.Errorf("failed to add profile: %w", err)
 	}
 
+	err = profiles.GetKoanfConfig().SaveProfile(newProfileName, subKoanf)
```

Comment:
`DefaultMissingKoanfKeys` should handle save profile. See existing `DefaultMissingViperKeys` on main branch

Reply:
It does. I'll remove this unneeded call.